src/core/generator/from_proto.rs

```diff
@@ -223,11 +223,18 @@ fn graphql_type_from_ref(name: &str) -> Result<GraphQLType<Unparsed>> {
 fn convert_primitive_type(proto_ty: &str) -> String {
     let binding = proto_ty.to_lowercase();
     let proto_ty = binding.strip_prefix("type_").unwrap_or(proto_ty);
+    // use Int64Str and Uint64Str to represent 64bit integers as string by default
+    // it's how this values are represented in JSON by default in prost
+    // see tests in `protobuf::tests::scalars_proto_file`
     match proto_ty {
         "double" | "float" => "Float",
-        "int32" | "int64" | "fixed32" | "fixed64" | "uint32" | "uint64" => "Int",
+        "int32" | "sint32" | "fixed32" | "sfixed32" => "Int",
+        "int64" | "sint64" | "fixed64" | "sfixed64" => "Int64Str",
+        "uint32" => "UInt32",
+        "uint64" => "Uint64Str",
         "bool" => "Boolean",
-        "string" | "bytes" => "String",
+        "string" => "String",
+        "bytes" => "Bytes",
```

Comment:
I think for numbers we should use the following:

| Rust Types                     | Protobuf Primitive            | GraphQL Scalar   |
|--------------------------------|-------------------------------|------------------|
| `i8`, `i16`, `i32`             | `int32`                       | `Int`            |
| `u8`, `u16`, `u32`             | `uint32`                      | `Int`            |
| `i64`                          | `int64`                       | `BigInt`         |
| `u64`                          | `uint64`                      | `BigInt`         |
| `f32`                          | `float`                       | `Float`          |
| `f64`                          | `double`                      | `Float`          |
| `bool`                         | `bool`                        | `Boolean`        |
| `char`                         | `string`                      | `String`         |
| `usize`, `isize`               | `int64` or `uint64`           | `BigInt`         |

Generally the consumer of GraphQL is going to be a JS developer who doesn't work with so many types usually. It would be easier on them to reduce it to some simple one that are compatible with something that exists in JS. This helps build a better intuition for a developer who is consuming a GraphQL API. It's hard for a JS engineer to make sense of what should `UInt32` map to natively.

Reply:
I'm not sure that wrapping so many different types into single one is a robust solution since the conversion from BigInt to any other int type will lose the precision or fail. And for grpc case, API expects a value with defined restrictions that could be expressed with respective scalar type. In that case scalar acts as validation step before sending invalid data to the upstream or generating conversion error. Also it acts as documentation on type level that helps to understand API expectations without a need to search for API docs or explore validation errors.

I feel like having such validation as scalar type (if we improve error message) will be easier to understand compared to conversion error or API error that could appear in such cases.

Reply:
I think my approach to scalars is similar to graphql-scalars package https://the-guild.dev/graphql/scalars/docs#why whilst single `BigInt` type simplifies definitions but doesn't reflect actual API restrictions